src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,218 @@
+      const modifiedFeatureIds = event.features
+        .getArray()
+        .map(feature => getFeatureId(feature))
+        .filter(id => !!id);
+      if (modifiedFeatureIds.length) {
+        this.#modifiedFeatureIDs.push(...modifiedFeatureIds);
+        this.#olSourceLayer.changed();
+      }
+    });
+    this.modifyInteraction.on('modifyend', event => {
+      this.#handleClick(event);
+    });
+    this.drawInteraction.setActive(false);
+    this.modifyInteraction.setActive(false);
+  }
+
+  setMap(map) {
+    const currentMap = this.getMap();
+    if (currentMap) {
+      currentMap.removeLayer(this.#editLayer);
+    }
+    if (!map) {
+      return;
+    }
+    map.addLayer(this.#editLayer);
+    super.setMap(map);
+    this.modifyInteraction.setMap(this.#map);
+    this.drawInteraction.setMap(this.#map);
+  }
+
+  drawInteraction;
+
+  modifyInteraction;
+
+  /**
+   * @param {import('ol/MapBrowserEvent.js').default} event
+   */
+  #handleClick(event) {
+    this.#yioMap.dispatchEvent(new PointerEvent('click', event.originalEvent));
+  }
+
+  /**
+   * original style of the content layer
+   */
+  #originalStyle = null;
+
+  setActive(active) {
+    if (!this.modifyInteraction || !this.drawInteraction) {
+      return;
+    }
+    const sourceLayer = getOlLayerForMbLayer(
+      this.#yioMap._getContentLayer(),
+      this.#yioMap.editLayer,
+    );
+    if (active) {
+      this.#olSourceLayer = sourceLayer;
+      this.#originalStyle = sourceLayer.getStyle();
+      sourceLayer.setStyle((feature, resolution) => {
+        if (this.#modifiedFeatureIDs.includes(getFeatureId(feature))) {
+          return;
+        }
+        return this.#originalStyle(feature, resolution);
+      });
+    } else {
+      if (this.#originalStyle) {
+        this.#olSourceLayer.setStyle(this.#originalStyle);
+      }
+      this.#olSourceLayer = null;
+    }
+    this.modifyInteraction.setActive(!!active);
+    this.drawInteraction.setActive(!!active);
+    this.#editLayer.getSource().clear();
+    super.setActive(active);
+  }
+
+  /**
+   * sets the style of the editing layer to match the existing layer of the content map.
+   */
+  setStyle() {
+    this.#editLayer.setStyle((feature, resolution) => {
+      return (
+        this.#olSourceLayer.getStyleFunction()(feature, resolution) ||
+        defaultStyles
+      );
+    });
+  }
+
+  /**
+   * @param {import('ol/MapBrowserEvent.js').default} event
+   * @returns {boolean}
+   */
+  handleEvent(event) {
+    let propagateEvent = true;
+
+    if (event.type === 'click') {
+      const map = this.#yioMap._getMap();
+      const existingFeature = map
+        .getFeaturesAtPixel(event.pixel, {
+          hitTolerance: 7,
+        })
+        .find(f => {
+          return f.get('layer') === this.#yioMap.editLayer;
+        });
+
+      if (existingFeature && isPointFeature(existingFeature)) {
+        this.#modifiedFeatureIDs.push(getFeatureId(existingFeature));
```

Comment:
Avoiding duplicates is necessary here as well. I'd recommend an `addModifiedFeatureId()` method, which returns `false` when the feature is in the array already.